.github/workflows/ci.yaml

```diff
@@ -96,32 +98,140 @@ jobs:
+        # See: https://github.com/docker/build-push-action/issues/225#issuecomment-727639184
+        name: Push base Docker image
+        uses: docker/build-push-action@v4
+        if: ${{ (github.ref_name == 'main') || (github.event_name == 'release') }}
+        with:
+          context: .
+          tags: ${{ steps.meta.outputs.tags }}
+          push: true
+
+  # Build and Push the predefined BitOps Docker image(s)
+  build-prebuilt-images:
+    runs-on: ubuntu-latest
+    needs:
+      - build-base-image
+
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - aws-ansible
+          - aws-helm
+          - aws-terraform
+          - omnibus
 
-      - name: List Docker images
-        run: docker images
+    steps:
+      - 
+        uses: actions/checkout@v3
+      - 
+        name: Build FROM 'dev-base' Dockerfile
+        uses: cuchi/jinja2-action@v1.2.0
+        if: ${{ (github.ref_name == 'main') || (github.event_name == 'pull_request') }}
+        with:
+          template: prebuilt-config/dockerfile.template
+          output_file: ./Dockerfile
+          variables: tag=dev-base
+      - 
+        name: Build FROM '1.2.3-base' Dockerfile
+        uses: cuchi/jinja2-action@v1.2.0
+        if: github.event_name == 'release'
+        with:
+          template: prebuilt-config/dockerfile.template
+          output_file: ./Dockerfile
+          # tag=v1.2.3-base -> tag=1.2.3-base is converted in the Dockerfile.template Jinja
+          variables: tag=${{ github.event.release.tag_name }}-base
+      - 
+        name: Use ${{ matrix.target }} bitops.config.yaml
+        run: cp ./prebuilt-config/${{ matrix.target }}/bitops.config.yaml ./bitops.config.yaml
+      - 
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - 
+        name: Tag the Docker image
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            bitovi/bitops
+          # disable automatic 'latest' tag generation
+          flavor: latest=false
+          # https://bitops.sh/versioning/
+          tags: |
+            # DEVELOPMENT
+            # On 'main' branch push, - build 'omnibus' image and publish it as 'dev' Docker tag
+            type=raw,value=dev,enable=${{ (github.ref_name == 'main') && (matrix.target == 'omnibus') }}
+            # RELEASE
+            # On a versioned release push '2.0.0-omnibus', '2.0.0-aws-ansible', etc Docker tags
+            type=semver,pattern={{version}}-${{ matrix.target }},enable=${{ github.event_name == 'release' }}
+            # If omnibus release, push additional versioned '1.2.3' Docker tag
+            type=semver,pattern={{version}},enable=${{ (github.event_name == 'release') && (matrix.target == 'omnibus') }}
+            # On release update additional 'latest' Docker tag for Omnibus image
+            type=raw,value=latest,enable=${{ (github.event_name == 'release') && (matrix.target == 'omnibus') }}
```

Comment:
The BitOps versioning tagging strategy is happening here with help of
https://github.com/marketplace/actions/docker-metadata-action